File: Arrays/MoveZeroes/Python/283-Move-Zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        \\\
        Do not return anything, modify nums in-place instead.
        \\\
        nonZeroIdx = 0

        for i in range(len(nums)):
            if nums[i] != 0:
                nums[nonZeroIdx], nums[i] = nums[i], nums[nonZeroIdx]
                nonZeroIdx += 1

File: Arrays/MoveZeroes/Go/283-Move-Zeroes.go
func moveZeroes(nums []int)  {
    zeroIdx := 0

    for i := 0; i < len(nums); i ++ {
        if nums[i] != 0 {
            nums[zeroIdx], nums[i] = nums[i], nums[zeroIdx]
            zeroIdx += 1
        }
    }
    
}